extensions/amp-animation/0.1/install-polyfill.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Deferred} from '../../../src/utils/promise';
+import {Services} from '../../../src/services';
+
+let polyfillRequested = false;
+const polyfillPromise_ = new Deferred();
+
+/**
+ * @param {!Window} win
+ * @return {Promise<void>}
+ */
+export function installWebAnimationsIfNecessary(win) {
+  if (!polyfillRequested) {
```

Comment:
We might have an issue with FIE, because the polyfill is requested for a window, but the `polyfillRequested` and `polyfillPromise_` are global.

Reply:
Is there a test suite for the FIE integration with AMP Animations?

Reply:
I'm addressing this via a `Map` per `window` object. There is an existing test, but it doesn't cover the situation well enough (it still has a singleton usage).